File: LeetCode/Python/minStack.py
class MinStack:

    def __init__(self):
        """
        initialize your data structure here.
        """
        self.stack = []
        

    def push(self, x: int) -> None:
        currMin = self.getMin()
        if currMin is None or x < currMin:
            currMin = x
        self.stack.append((x, currMin))

    def pop(self) -> None:
        self.stack.pop()[0]

    def top(self) -> int:
        return self.stack[len(self.stack)-1][0]

    def getMin(self) -> int:
        if len(self.stack) == 0:
            return None
        else:
            return self.stack[len(self.stack) - 1][1]

File: LeetCode/C#/MinStack.cs
public class MinStack {
    private List<int[]> stack;
    /** initialize your data structure here. */
    public MinStack() {
        this.stack = new List<int[]>();
    }
    
    public void Push(int x) {
        int currentMin = GetMin();
        if(x < currentMin){
            currentMin = x;
        }
        this.stack.Add(new int[]{x, currentMin});
    }
    
    public void Pop() {
        this.stack.RemoveAt(this.stack.Count - 1);
    }
    
    public int Top() {
        return this.stack[this.stack.Count - 1][0];
    }
    
    public int GetMin() {
        if(this.stack.Count == 0) return Int32.MaxValue;
        return this.stack[this.stack.Count - 1][1];
    }
}

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack obj = new MinStack();
 * obj.Push(x);
 * obj.Pop();
 * int param_3 = obj.Top();
 * int param_4 = obj.GetMin();
 */